ensime_shared/client.py

```diff
@@ -593,38 +619,56 @@ def type_check(self, filename):
             {"typehint": "TypecheckFilesReq",
              "files": [self.editor.path()]})
 
-    def unqueue(self, timeout=10, should_wait=False):
+    def unqueue(self, timeout=10):
         """Unqueue all the received ensime responses for a given file."""
-        start, now = time.time(), time.time()
-        wait = self.queue.empty() and should_wait
-
-        while (not self.queue.empty() or wait) and (now - start) < timeout:
-            if wait and self.queue.empty():
-                time.sleep(0.25)
-                now = time.time()
-            else:
-                result = self.queue.get(False)
-                self.log.debug('unqueue: result received\n%s', result)
-                if result and result != "nil":
-                    wait = None
-                    # Restart timeout
-                    start, now = time.time(), time.time()
+        with catch(Exception, self.log.error):
+            if timeout != 0 or not self.queue.empty():
+                if timeout == 0:
+                    timeout = 0.01
```

Comment:
should probably reintroduce should_wait param... although... maybe another function might be better for that?